pkg/image/build.go

```diff
@@ -0,0 +1,48 @@
+package image
+
+import (
+	"context"
+	"log"
+	"os"
+	"time"
+
+	"github.com/google/go-containerregistry/pkg/authn"
+	"github.com/google/go-containerregistry/pkg/crane"
+	v1 "github.com/google/go-containerregistry/pkg/v1"
+	"github.com/google/go-containerregistry/pkg/v1/empty"
+	"github.com/google/go-containerregistry/pkg/v1/mutate"
+	"github.com/google/go-containerregistry/pkg/v1/tarball"
+
+	tar "kubevirt.io/containerdisks/pkg/build"
+)
+
+func Build(diskPath string) (v1.Image, error) {
+	layer, err := tarball.LayerFromOpener(tar.StreamLayerOpener(diskPath))
+	if err != nil {
+		log.Fatalf("Error creating layer from file: %v", err)
+		return nil, err
+	}
+
+	image, err := mutate.AppendLayers(empty.Image, layer)
+	if err != nil {
+		log.Fatalf("Error appending layer: %v", err)
+		return nil, err
+	}
+	return image, nil
+}
+
+func Push(image v1.Image, imageDestination string, pushTimeout int) error {
```

Comment:
rename this file to image.go, or put the Push function to push.go file

Reply:
I'd prefer to rename it, because creating a one file for single function might be an overkill.